### DISCUSSION TITLE
Dúvida modulo: Mão na massa: Cronômetro

### DISCUSSION THREAD

**Discussion by mayerbryan** [balta-io/2801#59] [Perguntas e Respostas] [answered]
Neste módulo nós usamos a função Thread.sleep para segurar o while.. eu programava na Unity e usavamos o deltaTime para verificar a passagem do tempo, e pesquisando no google vi que o C# tem a função stopwatch, então gostaria de saber pq usar o sleep ao invés destas outras opções?

obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @mayerbryan, como vai?

Foi simplesmente por didática, não teve um motivo especial :)

**Reply by mayerbryan** [reply]
ah entendi, muito obrigado Balta! eu perguntei por achar que precisava usar aquele tipo de método para esse tipo de situação, era só pra entender se tinha alguma regra. muito obrigado!

**Comment by mayerbryan** [comment]
não sei se posso fazer outra pergunta aqui dentro, mas caso tenha que abrir um novo tópico por favor me avise. neste mesmo módulo eu estava tentando fazer um cronometro que roda infinito até o usuário apertar alguma tecla para parar a execução. eu montei esse código:

            Console.Clear();
            Console.WriteLine("Press Enter to start");
            Console.ReadLine();
            int time = 0;
            while (!Console.KeyAvailable)
            {
                Console.Clear();
                time++;
                Console.WriteLine(time);
                Thread.Sleep(1000);
            }
            Console.Clear();
            Console.WriteLine("Final Time: "+ time);
            Console.WriteLine("Press Enter to go back to Menu");
            Console.ReadLine();
            Console.ReadLine();
            Menu(); 
            
            como pode ver no final tem o console.readline() 2 vezes mas se eu tiro esse segundo quando eu clico para parar o loop o código passa direto pelo restante do codigo e joga de volta ao menu, o que eu poderia fazer para exiber o tempo que foi cronometrado e esperar o input do usario para voltar ao menu sem ter que usar esse console.readline() duas vezes?
